### ISSUE TITLE
A little discrepancy of chat templates (bug?)

### ISSUE THREAD

**Issue by Veason-silverbullet** [2U1/Qwen2-VL-Finetune#75]
The official Qwen2-VL chat template is compiled as

`user_input = f"{DEFAULT_IM_START_TOKEN}{user_input['role']}\n{user_input['content']}{DEFAULT_IM_END_TOKEN}\n{DEFAULT_IM_START_TOKEN}{gpt_response['role']}\n"`

This repo use the chat template 

`user_input = f"{DEFAULT_IM_START_TOKEN}{user_input['role']}\n{user_input['content']}\n{DEFAULT_IM_END_TOKEN}\n{DEFAULT_IM_START_TOKEN}{gpt_response['role']}\n"`

This is a little inconsistent, maybe should consider fix it.

**Comment by 2U1** [comment]
I think the chat template has changed after I made this repo. I think it wouldn't affect the performance but, in case it could so, I've updated the chat teamplate part. 

Thanks for letting me know.

**Comment by Veason-silverbullet** [comment]
@2U1 Many thanks for your work.

Here, there may be another bug - the learning rate scheduler. The config sets the sheduler as `cosine with warm-up (0.03)`, however, it is actual a constant scheduler with warm-up (0.03).

![Image](https://github.com/user-attachments/assets/dac769fe-bfb4-42c2-9eca-6ddf2c1e9a5e)

**Comment by 2U1** [comment]
Thanks for the report.
Removing the schedular part would make the scheduling work again.

**Comment by shineYuSong** [comment]
I use the chat template too, and how to test the result? send the query together to the model at once?
